src/dials/array_family/flex_ext.py

```diff
@@ -1297,14 +1299,44 @@ def map_centroids_to_reciprocal_space(
         # reciprocal lattice viewer -> but if we are looking at the crystal
         # coordinate frame we need to look at the experiments independently
 
+        if hasattr(experiments, "detectors"):
+            detectors = experiments.detectors()
```

Comment:
I think this is excessively defensive:

```
Python 3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 08:55:37) 
[Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from dxtbx.model.experiment_list import ExperimentList
>>> el = ExperimentList()
>>> el.detectors()
[]
>>> hasattr(el, "detectors")
True
```

even an empty experiment list would pass that...